src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -211,6 +289,79 @@ private IFeatureFilterMetadata GetFeatureFilterMetadata(string filterName)
             return filter;
         }
 
+        private IFeatureVariantAssignerMetadata GetFeatureAssignerMetadata(string assignerName)
+        {
+            IFeatureVariantAssignerMetadata assigner = _assignerMetadataCache.GetOrAdd(
+                assignerName,
+                (_) => {
+
+                    IEnumerable<IFeatureVariantAssignerMetadata> matchingAssigners = _variantAssigners.Where(f =>
+                    {
+                        string name = GetMetadataName(f.GetType());
+
+                        return IsMatchingMetadataName(name, assignerName);
+                    });
+
+                    if (matchingAssigners.Count() > 1)
+                    {
+                        throw new FeatureManagementException(FeatureManagementError.AmbiguousFeatureFilter, $"Multiple feature assigners match the configured assigner named '{assignerName}'.");
+                    }
+
+                    return matchingAssigners.FirstOrDefault();
+                }
+            );
+
+            return assigner;
+        }
+
+        private static string GetMetadataName(Type type)
```

Comment:
Can we be explicit (add another parameter) about whether the caller is for the filter or the assigner, so we don't need to worry about an assigner may end up with a filter in name?

Reply:
I think that's a good idea.

Reply:
Added.